Updated: 20 Feb 2021 by Nicolas Lorin <androw95220@gmail.com>

From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 19 Aug 2021 23:08:13 +0200
Subject: Add suffix "-insecure" to library with MI_SECURE=OFF

We want to make the secure variant the default one.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,12 +66,7 @@
   endif()
 endif()
 
-if("${CMAKE_BINARY_DIR}" MATCHES ".*(S|s)ecure$")
-  message(STATUS "Default to secure build")
-  set(MI_SECURE "ON")
-endif()
 
-
 # -----------------------------------------------------------------------------
 # Process options
 # -----------------------------------------------------------------------------
@@ -290,16 +285,17 @@
   set(mi_install_cmakedir   "${CMAKE_INSTALL_LIBDIR}/cmake/mimalloc-${mi_version}") # for cmake package info
 endif()
 
-set(mi_basename "mimalloc")
 if(MI_SECURE)
-  set(mi_basename "${mi_basename}-secure")
+  set(mi_basename "mimalloc-secure")
+else()
+  set(mi_basename "mimalloc-insecure")
 endif()
 if(MI_VALGRIND)
   set(mi_basename "${mi_basename}-valgrind")
 endif()
 string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LC)
 if(NOT(CMAKE_BUILD_TYPE_LC MATCHES "^(release|relwithdebinfo|minsizerel|none)$"))
-  set(mi_basename "${mi_basename}-${CMAKE_BUILD_TYPE_LC}") #append build type (e.g. -debug) if not a release version
+  set(mi_basename "mimalloc-${CMAKE_BUILD_TYPE_LC}") #append build type (e.g. -debug) if not a release version
 endif()
 
 if(MI_BUILD_SHARED)
